### ISSUE TITLE
Livesplit lagging in GPU intensive areas in DOOM 2016

### ISSUE THREAD

**Issue by Extra-Ghey** [LiveSplit/LiveSplit#2217]
So for some reason this is not a common issue as far as I can tell, but for me I run into some pretty nasty lag with livesplit when running doom 2016. The game doesn't lag, but when I hit 99 percent GPU usage I can see on my second monitor that the timer has pretty much stopped. When I exit the GPU intensive area the timer returns to normal and there is no timer drift, Everything is still accurate. Basically if my FPS is dropping, the timer is lagging. I know from past recordings of me running the game that I DID have the game running at high graphics settings, and the GPU intensive areas were not a problem at all. At some point I changed the settings somewhere and never made a record of what I did to make the timer run correctly, despite constantly hitting 99% GPU usage. I know that for me I need to run livesplit at real time process priority to work at all, which is weird since I've never seen that discussed anywhere.

Basically I'm trying to figure out how tf livesplit even runs. According to the task manager it uses 0% of the GPU, and even at the intensive spots in DOOM I'm only using 40% of the CPU at most. I've tried setting affinity and running as admin, nothing seems to change.

Any insight to what to do with this mess would be greatly appreciated, I just want to get back into this game.

Cheers.

**Comment by Extra-Ghey** [comment]
I would like to continue with an update on the situation. So apparently the main issue as I've discovered has nothing to do with GPU or CPU or RAM usage, and everything to do with Doom 2016 running in fullscreen or borderless mode. I've tried disabling full screen optimizations with both Doom and livesplit and still, fullscreen Doom does not cooperate with livesplit. Running the game in window mode allows the timer to run perfectly with zero lag, but windowed mode is not an option for me as it adds input lag. I KNOW that in the past I was able to run Doom in fullscreen, use livesplit and everything was fine, so its not a question of whether or not it's possible, it's a question of HOW is it possible.

I'm trying hard to find what the conflict is between fullscreen Doom and livesplit, so if anyone has any idea what could be done I could kiss you, this is eating my sanity.

Cheers.

**Comment by mleise** [comment]
Hmm, would be interesting to see what happens if you have some other simple animation playing on the 2nd screen; not HW accelerated video, but maybe just a simple stop watch.

You know what else kicks in when you go fullscreen? GSYNC or FreeSync. I have a hunch that you are using one of these. In 2020 Microsoft updated their Desktop Window Manager (DWM) to refresh the desktop at the refresh rate of the fastest monitor connected to the system. (Windows 10 build 2004)

My suspicion is that - possibly in connection with that update -, as soon as Doom 2016 slows down and the gaming monitor drops to lower than desktop refresh rate, Windows doesn't get notified about the VSync any more, which in turn breaks the synchronized update of the secondary monitor.
